difficulty-levels

jk370 opened the pull request.

jk370 committed:
Minor grammar edit to splash screen commit and removed duplicate TODO
Refactoring of bot movement to reduce code reuse and stop bot from eating walls.

jk370 committed:
Added action dialogue into game, uses a separate panel with layout.
Changed some return types to String to print to action box.
Saved score when a life is lost.
Disabled the text areas so you cannot click off of the game.
Interface one line change to match change in return type for movePlayer.
Removed two magic numbers

jk370 committed:
Merge branches 'action-dialogue-box' and 'master' of https://github.com/theaaron123/dungeonsgame into action-dialogue-box

jk370 committed:
fixed error with starting location overwriting room/boundaries on odd occasions.
Chest now can mysteriously vanish on occasion.

jk370 committed:
Difficulty levels added using a static value assigned to Dungeon.
Number of chests and rooms increased in hard mode.
size of dungeon increased in hard mode.

theaaron123 commented on src/main/java/uk/ac/bath/se/DungeonController.java:
```diff
@@ -187,30 +205,46 @@ private void addRoomBounds(Dungeon dungeon, int roomQuantity) {
                     addGold(room, startHeight, startWidth);
                     break;
 
+                case 11:
```
Why empty cases out of order, can a default case capture this instead

jk370 replied:
the cases roll down to run the case below. This makes the rooms be put in the correct quadrant when there are more than four

theaaron123 replied:
Add a comment saying intentional fall through because its often a bug when cases dont break

theaaron123 replied:
https://google.github.io/styleguide/javaguide.html#s4.8.4-switch

theaaron123 commented on src/main/java/uk/ac/bath/se/DungeonController.java:
```diff
@@ -238,6 +272,17 @@ private void addRoomBounds(Dungeon dungeon, int roomQuantity) {
     }
 
+    private boolean roomCollision(int startHeight, int startWidth) {
+        for (int i = startHeight-1; i <= startHeight+5; i ++) {
```
What is 5

jk370 replied:
5 is the room height/width (loop is searching one layer around the outside of the room to allow room for passages)

theaaron123 replied:
Declare it as a variable with an easy to to understand name

jk370 committed:
switch fall through and room size

fall through comments added in switch statement for addRoomBounds
variable for room size declared in roomCollision
